### PR TITLE
When the edge node goes offline, add pod restart and status query functions for the edge node

### PR DESCRIPTION
**Pull request by luomengY**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubeedge/kubeedge/issues/5427

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/get/get.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package get
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/get/get.go:1]: Plz add KubeEdge License header for all new files.

**Comment by fisherxu** [comment]
Could you plz add some demo show for the new commands in PR description? Thanks

**Comment by luomengY** [comment]
> Could you plz add some demo show for the new commands in PR description? Thanks
In Proposal: https://github.com/kubeedge/kubeedge/pull/5535, a demo of the command has been written.
image

image

**Comment by luomengY** [comment]
> Could you plz add some demo show for the new commands in PR description? Thanks



> Could you plz add some demo show for the new commands in PR description? Thanks

https://github.com/kubeedge/kubeedge/pull/5535

**Review by fisherxu — COMMENTED** [review]
As discussed with @Shelley-BaoYue, keadm is usually used as a control plane tool, to install the cloudcore/edgecore  components.  But in this scenario, it's used in data plane.

`keadm get`  `keadm restart`, looks like they are not very clear what they will control, the components or the pods? So we can consider to add a subcommand for `data plane`, the name which can have more disscusion, like `app`, or any others  cc @WillardHu @luomengY @tangming1996

File: keadm/cmd/keadm/app/cmd/cmd_others.go [hunk 2]
```diff
@@ -91,5 +93,8 @@ func NewKubeedgeCommand() *cobra.Command {
 
 	cmds.AddCommand(edge.NewEdgeRollback())
 
+	cmds.AddCommand(get.NewEdgeGet())
```
Comment (WillardHu) [inline keadm/cmd/keadm/app/cmd/cmd_others.go:96]: Should we consider adding certs parent flags here for commands that the get pod and the restart pod?

**Comment by luomengY** [comment]
> As discussed with @Shelley-BaoYue, keadm is usually used as a control plane tool, to install the cloudcore/edgecore components. But in this scenario, it's used in data plane.
> 
> `keadm get` `keadm restart`, looks like they are not very clear what they will control, the components or the pods? So we can consider to add a subcommand for `data plane`, the name which can have more disscusion, like `app`, or any others cc @WillardHu @luomengY @tangming1996

Perhaps we need to consider future expansion features and see everyone's opinions.

**Comment by WillardHu** [comment]
> As discussed with @Shelley-BaoYue, keadm is usually used as a control plane tool, to install the cloudcore/edgecore components. But in this scenario, it's used in data plane.
`keadm get` `keadm restart`, looks like they are not very clear what they will control, the components or the pods? So we can consider to add a subcommand for `data plane`, the name which can have more disscusion, like `app`, or any others cc @WillardHu @luomengY @tangming1996

We can create an issue to discuss the name of the tool.

**Comment by luomengY** [comment]
> > As discussed with @Shelley-BaoYue, keadm is usually used as a control plane tool, to install the cloudcore/edgecore components. But in this scenario, it's used in data plane.
> > `keadm get` `keadm restart`, looks like they are not very clear what they will control, the components or the pods? So we can consider to add a subcommand for `data plane`, the name which can have more disscusion, like `app`, or any others cc @WillardHu @luomengY @tangming1996
> 
> We can create an issue to discuss the name of the tool.

ok

File: keadm/cmd/keadm/app/cmd/ctl/ctl.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/ctl/get"
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/ctl/restart"
+)
+
+var (
+	ctlLongDescription  = `"keadm ctl" command operates on the data surface of edge nodes`
+	ctlShortDescription = `ctl function to operate on the data surface of edge nodes`
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/ctl/ctl.go:12]:
```suggestion
	ctlShortDescription = `Commands operating on the data plane at edge`
```
We can use the same for Long and Short?

File: keadm/cmd/keadm/app/cmd/ctl/ctl.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package ctl
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/ctl/ctl.go:1]: Lincense header

File: keadm/cmd/keadm/app/cmd/ctl/ctl.go [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/ctl/restart"
+)
+
+var (
+	ctlLongDescription  = `"keadm ctl" command operates on the data surface of edge nodes`
+	ctlShortDescription = `ctl function to operate on the data surface of edge nodes`
+)
+
+// NewCtlGet returns KubeEdge edge pod command.
+func NewCtlGet() *cobra.Command {
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/ctl/ctl.go:16]:
```suggestion
func NewCtl() *cobra.Command {
```

File: keadm/cmd/keadm/app/cmd/ctl/get/get.go [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+package get
+
+import "github.com/spf13/cobra"
+
+var (
+	edgeGetLongDescription  = `"keadm get" command get resources in edge node`
+	edgeGetShortDescription = `get function to get resources in edge node`
+)
+
+// NewEdgeGet returns KubeEdge edge pod command.
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/ctl/get/get.go:26]:
```suggestion
// NewEdgeGet returns KubeEdge edge resources get command.
```

File: keadm/cmd/keadm/app/cmd/ctl/get/pod.go [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+	"k8s.io/kubernetes/pkg/printers/storage"
+
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/common"
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/ctl/restful"
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/util"
+)
+
+var (
+	edgePodGetLongDescription  = `"keadm ctl get pod" command get pods in edge node`
+	edgePodGetShortDescription = `ctl get pod function to get pods in edge node`
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/ctl/get/pod.go:42]:
```suggestion
	edgePodGetShortDescription = `Get pods in edge node`
```

File: keadm/cmd/keadm/app/cmd/ctl/restart/pod.go [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/util"
+)
+
+type PodRestartOptions struct {
+	Namespace string
+}
+
+var (
+	edgePodRestartLongDescription  = `"keadm restart pod" command delete pods in edge node`
+	edgePodRestartShortDescription = `restart function to restart pods in edge node`
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/ctl/restart/pod.go:41]:
```suggestion
	edgePodRestartShortDescription = `Restart pods in edge node`
```

File: keadm/cmd/keadm/app/cmd/ctl/restart/restart.go [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+limitations under the License.
+*/
+
+package restart
+
+import "github.com/spf13/cobra"
+
+var (
+	edgeRestartLongDescription  = `"keadm ctl restart" command delete resources in edge node`
+	edgeRestartShortDescription = `restart ctl function to restart resources in edge node`
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/ctl/restart/restart.go:23]:
```suggestion
	edgeRestartShortDescription = `Restart resources in edge node`
```
Should not delete at edge

**Review by fisherxu — COMMENTED** [review]
/approve
/assign @WillardHu @Shelley-BaoYue for final review :)

**Pull request merged**